addons/l10n_account_withholding/wizards/account_payment_register.py

```diff
@@ -0,0 +1,275 @@
+                        # We update the context of the base_line's tax_ids to trigger computation of the withholding taxes.
+                        base_line = line.move_id._prepare_product_base_line_for_taxes_computation(line)
+                        base_line['tax_ids'] = base_line['tax_ids'].with_context(include_withholding_taxes=True)
+                        self.env['account.tax']._add_tax_details_in_base_line(base_line, line.company_id)
+
+                        base_account = line.company_id.withholding_tax_base_account_id or line.account_id
+
+                        # We want to generate one line per group. A group is defined by a tax, account, and an analytic distribution.
+                        for withholding_tax in withholding_taxes:
+                            base_amount = 0
+                            if tax_data := [d for d in base_line['tax_details']['taxes_data'] if d['tax'] == withholding_tax][:1]:
+                                base_amount = tax_data[0]['raw_base_amount']  # We want the amount in company currency
+                            # Check if the move has a fiscal position and apply if needed.
+                            if move.fiscal_position_id:
+                                withholding_tax = move.fiscal_position_id.map_tax(withholding_tax)
+                            withholding_line_amounts[(json.dumps(base_line['analytic_distribution']), base_account, withholding_tax)] += base_amount
+
+                for (analytic_distribution, base_account, withholding_tax), withholding_line_amount in withholding_line_amounts.items():
+                    withholding_line_creation_vals.append(Command.create({
+                        'tax_id': withholding_tax.id,
+                        'analytic_distribution': json.loads(analytic_distribution),
+                        'original_base_amount': withholding_line_amount,
+                        'tax_base_account_id': base_account.id,
+                    }))
+            wizard.withholding_line_ids = withholding_line_creation_vals
+
+    @api.depends('withholding_line_ids')
+    def _compute_withhold_tax(self):
+        """ By default, we display the table only if we have default withholding taxes on any products. """
+        for wizard in self:
+            wizard.withhold_tax = bool(wizard.withholding_line_ids)
+
+    @api.depends('batches')
+    def _compute_payment_move_amount_total(self):
+        """ Get the full amount of the move related to this wizard.
+        We do not use the source_amount as it only takes into account residual amounts.
+        """
+        def get_total_in_company_currency(move):
+            total = sum(line.balance for line in move.line_ids if line.display_type in ('tax', 'product', 'rounding'))
+            return move.direction_sign * total
+
+        for wizard in self:
+            if wizard.country_code == 'AR' or not wizard.can_edit_wizard:
```

Comment:
gné? Not really really generic^^

Reply:
Argentina already has a withholding tax module with country-specific things in it.
Until we reach Master and clean things up so that the AR module only contains the country-specific logic and uses our system as the core, it is best to allow both modules to co-exist (could be needed for multiple companies in multiple countries) but to avoid impacting each other.

This module is for everything that needs it (but not AR), and the AR one is for AR only.